oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoVersionGCSupport.java

```diff
@@ -373,8 +368,8 @@ public void apply(BasicDBObject document) {
      */
     @Override
     public Optional<NodeDocument> getOldestModifiedDoc(final Clock clock) {
-        final Bson sort = and(eq(MODIFIED_IN_SECS, 1), eq(ID, 1));
-
+        final Bson sort = Sorts.ascending(MODIFIED_IN_SECS, ID);
```

Comment:
Could you please create a JIRA issue for this bug and submit a separate Pull Request? I'd like to separate bug fixes from other changes.

Reply:
Sorry @reschke could you guide me in doing what you are asking?
I've created the [new issue 11169](https://issues.apache.org/jira/browse/OAK-11169), now how can I submit new pull request while there is already this open? Should I create a new branch? In current pull-request should I rollback the fix in MongoVersionGCSupport leaving the code as I found initially?

Reply:
I would recommend that while the biggger change gets more testing, we treat the query bug as seperate issue and fix it, "just because". Once the fix is in, we can rebase the PR (or merge), and can proceed with this one.

@rishabhdaim ?